lnode/synctxnpool.go

```diff
@@ -0,0 +1,396 @@
+				log.Warningf("sendReqTxnPoolHash neighbor %v error: %v", nbr.GetAddr(), err)
+				return
+			}
+			log.Infof("Get reply TxnPoolHashAndCount, neighbor's txn count is: %v, hash is: %v, my hash: %v ",
+				replyMsg.TxnCount, hex.EncodeToString(replyMsg.PoolHash)[:8], hex.EncodeToString(myHash)[:8])
+
+			if !bytes.Equal(replyMsg.PoolHash, myHash) {
+				if replyMsg.TxnCount > maxCount { // Choose the neighbor which has most txn count in txn pool
+					neighborToSync = nbr
+					maxCount = replyMsg.TxnCount
+				}
+			}
+		}(neighbor)
+
+	}
+
+	wg.Wait()
+
+	// request for syncing
+	if neighborToSync != nil {
+		replyMsg, err := localNode.requestSyncTxnPool(neighborToSync)
+		if err != nil {
+			log.Error("req sync txn pool error: ", err)
+		} else {
+			// sort it by nonce to append them to txn pool.
+			sort.Slice(replyMsg.Transactions, func(i, j int) bool {
+				return replyMsg.Transactions[i].UnsignedTx.Nonce < replyMsg.Transactions[j].UnsignedTx.Nonce
+			})
+
+			for _, pbTxn := range replyMsg.Transactions {
+				txn := &transaction.Transaction{
+					Transaction: pbTxn,
+				}
+
+				localNode.TxnPool.AppendTxnPool(txn)
+			}
+
+			// after update my txn pool, reset sync time.
+			localNode.TxnPool.SetSyncTime()
+		}
+	}
+}
+
+// send neighbor a REQ_TXN_POOL_HASH request message, and wait for reply.
+func (localNode *LocalNode) requestTxnPoolHash(remoteNode *node.RemoteNode) (*pb.TxnPoolHashAndCount, error) {
+
+	msg, err := NewReqTxnPoolHashMessage()
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := localNode.SerializeMessage(msg, false)
+	if err != nil {
+		return nil, err
+	}
+
+	replyBytes, err := remoteNode.SendBytesSyncWithTimeout(buf, syncReplyTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	replyMsg := &pb.TxnPoolHashAndCount{}
+	err = proto.Unmarshal(replyBytes, replyMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	return replyMsg, nil
+}
+
+// build a REQ_TXN_POOL_HASH request message
+func NewReqTxnPoolHashMessage() (*pb.UnsignedMessage, error) {
+
+	msg := &pb.UnsignedMessage{
+		MessageType: pb.MessageType_REQ_TXN_POOL_HASH,
+	}
+
+	return msg, nil
+
+}
+
+// build RPL_TXN_POOL_HASH reply message, reply with my txn pool's hash and txn count.
+func NewRplTxnPoolHashMessage(tp *pool.TxnPool) (*pb.UnsignedMessage, error) {
+
+	poolHash, txnCount := tp.GetTwtFingerprint()
+
+	msgBody := &pb.TxnPoolHashAndCount{
+		PoolHash: poolHash,
+		TxnCount: (uint32)(txnCount),
+	}
+
+	buf, err := proto.Marshal(msgBody)
+	if err != nil {
+		return nil, err
+	}
+
+	msg := &pb.UnsignedMessage{
+		MessageType: pb.MessageType_RPL_TXN_POOL_HASH,
+		Message:     buf,
+	}
+
+	return msg, nil
+}
+
+// build REQ_SYNC_TXN_POOL request message. This message includse my txn pool's txn address and maximum nonce.
+func NewReqSyncTxnPoolMessage(tp *pool.TxnPool) (*pb.UnsignedMessage, error) {
+
+	addrNonceList := make([]*pb.AddrNonce, 0)
+	mapAddrNonce := make(map[common.Uint160]uint64)
+
+	// remove time over txn first
+	tp.RemoveExpiredTwt()
```

Comment:
Do NOT do such kind of hidden logic in constructor

Reply:
yes. removed.